.github/workflows/static-checks.yml

```diff
@@ -0,0 +1,76 @@
+name: Static checks and analysis
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  syntax-check:
+    runs-on: ubuntu-16.04
```

Comment:
It would be nice to run this test on `ubuntu-latest` as well. Are there any difficulties with latest PHP?

Reply:
The project is based on PHP 7.x and by default `ubuntu-latest` picks PHP 8.x. There have been some major changes(syntax) in PHP 8.0. We can try a workaround by forcing the Action to install PHP 7.x but it doesn't solve any purpose. The project will still be outdated in terms of PHP. 

Changes in PHP 8.0: https://www.php.net/manual/en/migration80.incompatible.php

You can see details of the test run here : https://github.com/avinal/fossology/runs/2097843501?check_suite_focus=true#step:4:206

I would suggest, we should migrate to PHP 8.x, can't say if this is desirable or not.

Reply:
It will be big of a task to make FOSSology PHP 8 compatible. It will require big effort on migration and testing with 7.0 and 8.x. Lets keep it for a separate PR.

As per the documentation, Ubuntu 20.04 image comes with PHP 7.4 and 8.0: https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#php

Can you please check if you can use the `update-alternatives` to update php to point to php-7.4?

Also, now that I think about it, changing the tag from `ubuntu-latest` to `ubuntu-20.04` will be better making sure the pipeline does not break spontaneously in future. What are your thoughts on it?

Reply:
May I open an issue for PHP upgrade??

I tried using `update-alternatives`, it is working on my local machine but not on GitHub Actions for some reasons. I will try once again with some modifications and let you know.

Since LTS versions are released after every two years, `ubuntu-latest` doesn't changes that often, but yeah for assurance `ubuntu-20.04` is surely a better choice. I will update that.

Reply:
Great, It worked, I might have missed something while testing :)